packages/plugin-ext/src/plugin/plugin-storage.ts

```diff
@@ -38,6 +38,10 @@ export class Memento implements theia.Memento {
         }
     }
 
+    keys(): string[] {
+        return Object.entries(this.cache ?? {}).filter(([, value]) => value !== undefined).map(([key]) => key);
```

Comment:
Question: I'm not too familiar with this code, but it seems like `this.cache` can never be undefined, right? Or are there cases in which the `?? {}` actually does something? If so, we should probably align `get<T>` and `update` so they also correctly deal with `cache` being undefined.

Reply:
You're right, we always make sure to properly initialize the cache so it is never `undefined`. 
I've updated the pull-request.